source/gitlab/gitlab.go

```diff
@@ -32,17 +31,22 @@ var (
 )
 
 type Gitlab struct {
+	config *Config
 	client *gitlab.Client
 	url    string
 
-	projectID   string
-	path        string
 	listOptions *gitlab.ListTreeOptions
 	getOptions  *gitlab.GetFileOptions
 	migrations  *source.Migrations
 }
 
 type Config struct {
+	Scheme    string
```

Comment:
Does the scheme need to be specified? When would it not be `https`?

Reply:
For local (in-house) Gitlab instances it's up to you to choose a scheme. Our internal network doesn't provide https access to Gitlab, only http.

Reply:
When you set the Scheme via URL e.g. in `Open()` you can't use `u.Scheme` since that'll be `gitlab`, not `http` or `https`. The scheme will need to be provided via a `x-scheme` URL query parameter.